autoreject/autoreject.py

```diff
@@ -466,58 +441,34 @@ def _vote_epochs(self, epochs):
 
-        self._drop_log = np.zeros((n_epochs, len(epochs.ch_names)))
-        self.bad_epoch_counts = np.zeros((len(epochs), ))
+        drop_log = np.zeros((n_epochs, len(epochs.ch_names)))
+        bad_sensor_counts = np.zeros((len(epochs), ))
 
         ch_names = [epochs.ch_names[p] for p in picks]
         deltas = np.ptp(epochs.get_data()[:, picks], axis=-1).T
         threshes = [self.threshes_[ch_name] for ch_name in ch_names]
         for ch_idx, (delta, thresh) in enumerate(zip(deltas, threshes)):
             bad_epochs_idx = np.where(delta > thresh)[0]
             # TODO: combine for different ch types
-            self.bad_epoch_counts[bad_epochs_idx] += 1
-            self._drop_log[bad_epochs_idx, picks[ch_idx]] = 1
+            bad_sensor_counts[bad_epochs_idx] += 1
+            drop_log[bad_epochs_idx, picks[ch_idx]] = 1
+        return drop_log, bad_sensor_counts
 
-    def _get_bad_epochs(self):
-        """Get the indices of bad epochs.
-        """
-        # TODO: this must be done separately for each channel type?
-        self.sorted_epoch_idx = np.argsort(self.bad_epoch_counts)[::-1]
-        bad_epoch_counts = np.sort(self.bad_epoch_counts)[::-1]
-        n_channels = len(self.picks)
-        n_consensus = self.consensus_perc * n_channels
-        if np.max(bad_epoch_counts) >= n_consensus:
-            self.n_epochs_drop = np.sum(self.bad_epoch_counts >=
-                                        n_consensus) + 1
-            bad_epochs_idx = self.sorted_epoch_idx[:self.n_epochs_drop]
-        else:
-            self.n_epochs_drop = 0
-            bad_epochs_idx = []
-
-        return bad_epochs_idx
-
-    def _interpolate_bad_epochs(self, epochs, verbose='progressbar'):
-        """interpolate the bad epochs.
-
-        Parameters
-        ----------
-        epochs : instance of mne.Epochs
-            The epochs object which must be fixed.
-        """
-        drop_log = self._drop_log
+    def _get_epochs_interpolation(self, epochs, drop_log,
+                                  ch_type, verbose='progressbar'):
+        """Interpolate the bad epochs."""
         # 1: bad segment, # 2: interpolated
-        self.fix_log = self._drop_log.copy()
+        fix_log = drop_log.copy()
         ch_names = epochs.ch_names
         non_picks = np.setdiff1d(range(epochs.info['nchan']), self.picks)
-        # TODO: raise error if preload is not True
-        pos = 4 if hasattr(self, '_leave') else 2
-        for epoch_idx in _pbar(range(len(epochs)), desc='Repairing epochs',
-                               position=pos, leave=True, verbose=verbose):
+        bad_channels = list()
+        n_interpolate = self.n_interpolate[ch_type]
+        for epoch_idx in range(len(epochs)):
             n_bads = drop_log[epoch_idx].sum()
             if n_bads == 0:
                 continue
             else:
-                if n_bads <= self.n_interpolate:
+                if n_bads <= n_interpolate:
```

Comment:
I think it is effectively correct because the drop_log gets recomputed each time.